### PR TITLE
Update to latest openethereum version

### PR DESCRIPTION
**Pull request by karlb**
The openethereum project removed the 3.0.1 release due to known problems with it. So we need to update to keep the CI running.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -142,7 +142,7 @@ commands:
             echo ${PARITY_VERSION} > /tmp/parity-version
             echo ${SOLC_VERSION} > /tmp/solc-version
       - restore_cache:
-          key: system-deps-v7-{{ arch }}-geth-{{ checksum "/tmp/geth-version" }}-parity-{{ checksum "/tmp/parity-version" }}-solc-{{ checksum "/tmp/solc-version" }}
+          key: system-deps-v8-{{ arch }}-geth-{{ checksum "/tmp/geth-version" }}-parity-{{ checksum "/tmp/parity-version" }}-solc-{{ checksum "/tmp/solc-version" }}
```
Comment (ulope) [inline .circleci/config.yml:145]: The checksums should automatically cause a new cache to be used.
It shouldn't be necessary to increase the version manually, that's only there as a fallback when the cache gets corrupted for some reason

**Review by ulope — APPROVED** [review]

File: raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py [hunk 2]
```diff
@@ -238,8 +238,8 @@ def test_remote_transaction_with_zero_gasprice_is_not_mined(
     except TransactionNotFound:
         zerogas_receipt = None
 
-    msg = "The transaction does NOT have a receipt because the miner rejected it"
-    assert zerogas_receipt is None, msg
```
Comment (ulope) [inline raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py:242]: Looks like the tests fail here:
```
assert zerogas_receipt["blockHash"] is None, msg
E       TypeError: 'NoneType' object is not subscriptable
```

File: raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py [hunk 3]
```diff
@@ -238,8 +238,8 @@ def test_remote_transaction_with_zero_gasprice_is_not_mined(
...
+    msg = "The transaction receipt does NOT have a blockHash because the miner rejected it"
+    assert zerogas_receipt["blockHash"] is None, msg
```
Comment (fredo) [inline raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py:242]:
```suggestion
    assert zerogas_receipt and zerogas_receipt["blockHash"] is None, msg
```

File: .circleci/config.yml (hunk 1 above)
Comment (karlb) [reply .circleci/config.yml:145]: I accidentally changes the solc link, which messed up the cache. A new checksum would have only been created if I changed one of the software version numbers, but all of these were already correct, so I increased the `v` value in the key.

File: raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py (hunk 3 above)
Comment (karlb) [reply raiden/tests/integration/rpc/assumptions/test_rpc_gas_price_assumptions.py]: Yes, that's what you get for testing with only one eth client locally.

**Pull request merged**